tools/src/main/java/org/apache/kafka/tools/TransactionalMessageCopier.java

```diff
@@ -137,16 +156,28 @@ private static ArgumentParser argParser() {
                 .dest("enableRandomAborts")
                 .help("Whether or not to enable random transaction aborts (for system testing)");
 
+        parser.addArgument("--group-mode")
+                .action(storeTrue())
+                .type(Boolean.class)
+                .metavar("GROUP-MODE")
+                .dest("groupMode")
+                .help("Whether to let consumer subscribe to the input topic or do manual assign. If we do" +
+                          " subscription based consumption, the input partition shall be ignored");
+
+        parser.addArgument("--use-group-metadata")
+                .action(storeTrue())
+                .type(Boolean.class)
+                .metavar("USE-GROUP-METADATA")
+                .dest("useGroupMetadata")
+                .help("Whether to use the new transactional commit API with group metadata");
+
         return parser;
     }
 
     private static KafkaProducer<String, String> createProducer(Namespace parsedArgs) {
-        String transactionalId = parsedArgs.getString("transactionalId");
-        String brokerList = parsedArgs.getString("brokerList");
-
         Properties props = new Properties();
-        props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, brokerList);
-        props.put(ProducerConfig.TRANSACTIONAL_ID_CONFIG, transactionalId);
+        props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, parsedArgs.getString("brokerList"));
+        props.put(ProducerConfig.TRANSACTIONAL_ID_CONFIG, parsedArgs.getString("transactionalId"));
```

Comment:
If we are in `groupMode` and enable `userGroupMetadata` should we use a different `transaction.id` for each producer instead of the same?

Reply:
Should be guaranteed by the python test suite level